src/components/scenes/FioRequestListScene.js

```diff
@@ -0,0 +1,481 @@
+                <View>{this.requestedTimeAndMemo(new Date(transaction.item.time_stamp), transaction.item.content.memo)}</View>
+              </View>
+            </View>
+            <View style={requestListStyles.columnCurrency}>
+              <View>{this.currencyField(transaction.item.content.token_code, transaction.item.content.amount)}</View>
+              <View>{this.fiatField(transaction.item.content.token_code, this.props.fiatSymbol, transaction.item.content.amount)}</View>
+            </View>
+          </View>
+        </View>
+      </TouchableHighlight>
+    )
+  }
+
+  renderSentTx = (transaction: Object) => {
+    const textStyles = this.stylesByStatus(transaction.item.status)
+    return (
+      <TouchableHighlight
+        onPress={() => this.selectSentRequest(transaction.item)}
+        style={
+          transaction.index === 0 ||
+          (transaction.index > 0 &&
+            this.dateOnly(new Date(this.props.sentFioRequests[transaction.index - 1].time_stamp)) !== this.dateOnly(new Date(transaction.item.time_stamp)))
+            ? requestListStyles.rowFrontWithHeader
+            : requestListStyles.rowFront
+        }
+        underlayColor={this.underlayColor}
+      >
+        <View key={transaction.item.fio_request_id.toString()}>
+          {transaction.index === 0 && <View>{this.headerRow(new Date(transaction.item.time_stamp))}</View>}
+          {transaction.index > 0 &&
+            this.dateOnly(new Date(this.props.sentFioRequests[transaction.index - 1].time_stamp)) !== this.dateOnly(new Date(transaction.item.time_stamp)) && (
+            <View>{this.headerRow(new Date(transaction.item.time_stamp))}</View>
+          )}
+          <View style={requestListStyles.rowItem}>
+            <View style={requestListStyles.columnItem}>
+              <View>{this.requestedIcon(transaction.item.status)}</View>
+              <View>
+                <View>{this.sentField(transaction.item.payer_fio_address)}</View>
+                <View>{this.requestedTimeAndMemo(new Date(transaction.item.time_stamp), transaction.item.content.memo)}</View>
+              </View>
+            </View>
+            <View style={requestListStyles.columnCurrency}>
+              <View>{this.currencyField(transaction.item.token_code, transaction.item.content.amount, textStyles)}</View>
+              <View>{this.fiatField(transaction.item.content.token_code, this.props.fiatSymbol, transaction.item.content.amount, textStyles)}</View>
+              <View>{this.statusField(transaction.item.status)}</View>
+            </View>
+          </View>
+        </View>
+      </TouchableHighlight>
+    )
+  }
+
+  render () {
+    const sentTxs = this.sortDescending(this.props.sentFioRequests)
+    const isAndroid = Platform.OS === 'android'
+    const { rejectLoading } = this.state
+    const { loading } = this.props
+    return (
+      <SceneWrapper>
+        {(rejectLoading || loading) && <FullScreenLoader />}
```

Comment:
Here is where you can conditionally render a full screen spinner and simply not render anything else if you are still loading

Reply:
why we should do this?
also this loader is used when user is rejecting request